lib/infra/TranscodingEngine.js

```diff
@@ -0,0 +1,64 @@
+
+const Promise = require("bluebird");
+const exec = require('child_process').exec;
+const TranscodingEngineResponse = require('./TranscodingEngineResponse');
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * class to handle transcoding
+ */
+class TranscodingEngine
+{
+
+    constructor(ffmpegPath = 'ffmpeg')
+    {
+        this._ffmpegPath=ffmpegPath;
+    }
+
+    static validateFilePaths(sourceFilePath , pathToSave)
+    {
+        return new Promise(function(resolve,reject) {
+            fs.exists(sourceFilePath, function (exists) {
+                if (exists)
+                {
+                    let $dirPathToSave = path.dirname(pathToSave);
+                    fs.exists($dirPathToSave, function (exists) {
+                        if (exists)
+                            resolve(true);
+                        else
+                            reject('[' + $dirPathToSave + '] doesn\'t exists on the file system');
+                    });
+                }
+                else
+                    reject('File [' + sourceFilePath + '] doesn\'t exists on the file system');
+            });
+        });
+    }
+
+    transcodeFile(commandLine)
+    {
+        let This = this;

```

Comment:
let means that you are going to reassign this variable - use const instead since you don't re-assign it